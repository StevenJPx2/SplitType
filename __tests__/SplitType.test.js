import SplitType from '../lib/SplitType'
import createElement from '../lib/utils/createElement'

describe('SplitType', () => {
  const elements = [
    createElement('div', { textContent: 'One Two', class: 'text' }),
    createElement('div', { textContent: 'Three Four', class: 'text' }),
  ]
  elements.forEach((element) => document.body.appendChild(element))

  it('Matches snapshot with default options', () => {
    const text = new SplitType('.text')
    expect(text).toMatchSnapshot()
  })
  it(`Matches snapshot with { types: "words, chars" }`, () => {
    const text = new SplitType('.text', { types: 'words, chars' })
    expect(text).toMatchSnapshot()
  })
})
