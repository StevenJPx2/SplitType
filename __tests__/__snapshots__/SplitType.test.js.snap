// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitType Matches snapshot with { types: "words, chars" } 1`] = `
SplitType {
  "chars": Array [
    <div
      class="char"
      style="display: inline-block;"
    >
      O
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      n
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      e
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      T
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      w
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      o
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      T
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      h
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      r
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      e
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      e
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      F
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      o
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      u
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      r
    </div>,
  ],
  "elements": Array [
    <div
      class="text"
    >
      <div
        class="word"
        style="display: inline-block; position: relative"
      >
        <div
          class="char"
          style="display: inline-block;"
        >
          O
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          n
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          e
        </div>
      </div>
       
      <div
        class="word"
        style="display: inline-block; position: relative"
      >
        <div
          class="char"
          style="display: inline-block;"
        >
          T
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          w
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          o
        </div>
      </div>
    </div>,
    <div
      class="text"
    >
      <div
        class="word"
        style="display: inline-block; position: relative"
      >
        <div
          class="char"
          style="display: inline-block;"
        >
          T
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          h
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          r
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          e
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          e
        </div>
      </div>
       
      <div
        class="word"
        style="display: inline-block; position: relative"
      >
        <div
          class="char"
          style="display: inline-block;"
        >
          F
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          o
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          u
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          r
        </div>
      </div>
    </div>,
  ],
  "isSplit": true,
  "lines": Array [],
  "originals": Array [
    "One Two",
    "Three Four",
  ],
  "settings": Object {
    "absolute": false,
    "charClass": "char",
    "lineClass": "line",
    "splitClass": "",
    "tagName": "div",
    "types": "words, chars",
    "wordClass": "word",
  },
  "words": Array [
    <div
      class="word"
      style="display: inline-block; position: relative"
    >
      <div
        class="char"
        style="display: inline-block;"
      >
        O
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        n
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        e
      </div>
    </div>,
    <div
      class="word"
      style="display: inline-block; position: relative"
    >
      <div
        class="char"
        style="display: inline-block;"
      >
        T
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        w
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        o
      </div>
    </div>,
    <div
      class="word"
      style="display: inline-block; position: relative"
    >
      <div
        class="char"
        style="display: inline-block;"
      >
        T
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        h
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        r
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        e
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        e
      </div>
    </div>,
    <div
      class="word"
      style="display: inline-block; position: relative"
    >
      <div
        class="char"
        style="display: inline-block;"
      >
        F
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        o
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        u
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        r
      </div>
    </div>,
  ],
}
`;

exports[`SplitType Matches snapshot with default options 1`] = `
SplitType {
  "chars": Array [
    <div
      class="char"
      style="display: inline-block;"
    >
      O
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      n
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      e
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      T
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      w
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      o
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      T
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      h
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      r
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      e
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      e
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      F
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      o
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      u
    </div>,
    <div
      class="char"
      style="display: inline-block;"
    >
      r
    </div>,
  ],
  "elements": Array [
    <div
      class="text"
    >
      <div
        class="line"
        style="display: block; text-align: ; width: 100%;"
      >
        <div
          class="word"
          style="display: inline-block; position: relative"
        >
          <div
            class="char"
            style="display: inline-block;"
          >
            O
          </div>
          <div
            class="char"
            style="display: inline-block;"
          >
            n
          </div>
          <div
            class="char"
            style="display: inline-block;"
          >
            e
          </div>
        </div>
         
        <div
          class="word"
          style="display: inline-block; position: relative"
        >
          <div
            class="char"
            style="display: inline-block;"
          >
            T
          </div>
          <div
            class="char"
            style="display: inline-block;"
          >
            w
          </div>
          <div
            class="char"
            style="display: inline-block;"
          >
            o
          </div>
        </div>
      </div>
    </div>,
    <div
      class="text"
    >
      <div
        class="line"
        style="display: block; text-align: ; width: 100%;"
      >
        <div
          class="word"
          style="display: inline-block; position: relative"
        >
          <div
            class="char"
            style="display: inline-block;"
          >
            T
          </div>
          <div
            class="char"
            style="display: inline-block;"
          >
            h
          </div>
          <div
            class="char"
            style="display: inline-block;"
          >
            r
          </div>
          <div
            class="char"
            style="display: inline-block;"
          >
            e
          </div>
          <div
            class="char"
            style="display: inline-block;"
          >
            e
          </div>
        </div>
         
        <div
          class="word"
          style="display: inline-block; position: relative"
        >
          <div
            class="char"
            style="display: inline-block;"
          >
            F
          </div>
          <div
            class="char"
            style="display: inline-block;"
          >
            o
          </div>
          <div
            class="char"
            style="display: inline-block;"
          >
            u
          </div>
          <div
            class="char"
            style="display: inline-block;"
          >
            r
          </div>
        </div>
      </div>
    </div>,
  ],
  "isSplit": true,
  "lines": Array [
    <div
      class="line"
      style="display: block; text-align: ; width: 100%;"
    >
      <div
        class="word"
        style="display: inline-block; position: relative"
      >
        <div
          class="char"
          style="display: inline-block;"
        >
          O
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          n
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          e
        </div>
      </div>
       
      <div
        class="word"
        style="display: inline-block; position: relative"
      >
        <div
          class="char"
          style="display: inline-block;"
        >
          T
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          w
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          o
        </div>
      </div>
    </div>,
    <div
      class="line"
      style="display: block; text-align: ; width: 100%;"
    >
      <div
        class="word"
        style="display: inline-block; position: relative"
      >
        <div
          class="char"
          style="display: inline-block;"
        >
          T
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          h
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          r
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          e
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          e
        </div>
      </div>
       
      <div
        class="word"
        style="display: inline-block; position: relative"
      >
        <div
          class="char"
          style="display: inline-block;"
        >
          F
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          o
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          u
        </div>
        <div
          class="char"
          style="display: inline-block;"
        >
          r
        </div>
      </div>
    </div>,
  ],
  "originals": Array [
    "One Two",
    "Three Four",
  ],
  "settings": Object {
    "absolute": false,
    "charClass": "char",
    "lineClass": "line",
    "splitClass": "",
    "tagName": "div",
    "types": "lines, words, chars",
    "wordClass": "word",
  },
  "words": Array [
    <div
      class="word"
      style="display: inline-block; position: relative"
    >
      <div
        class="char"
        style="display: inline-block;"
      >
        O
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        n
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        e
      </div>
    </div>,
    <div
      class="word"
      style="display: inline-block; position: relative"
    >
      <div
        class="char"
        style="display: inline-block;"
      >
        T
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        w
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        o
      </div>
    </div>,
    <div
      class="word"
      style="display: inline-block; position: relative"
    >
      <div
        class="char"
        style="display: inline-block;"
      >
        T
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        h
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        r
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        e
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        e
      </div>
    </div>,
    <div
      class="word"
      style="display: inline-block; position: relative"
    >
      <div
        class="char"
        style="display: inline-block;"
      >
        F
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        o
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        u
      </div>
      <div
        class="char"
        style="display: inline-block;"
      >
        r
      </div>
    </div>,
  ],
}
`;
